main.cpp

```diff
@@ -148,19 +146,19 @@ double Motorbike::getSpeed()
 }
 void Motorbike::calculateAndPrintRisk()
 {
-    for(size_t i = 0; i < passengers; ++i)
+    for(size_t i = 0; i < passengerRiskArray.size(); ++i)
     {
-        passengerRiskArray[i] = riskCoeffArray[i] * speed * passengers * PASSENGER_COEFF;
+        passengerRiskArray[i] = riskCoeffArray[i] * speed * passengerRiskArray.size() * PASSENGER_COEFF;
         std::cout << "Risk to passenger " << i + 1 << " is: " << passengerRiskArray[i] << " measured in fatalities per journey"<< std::endl;
     }
 }
 class Lorry
 {
 public:
     
-    Lorry(std::string name_, double speed_, unsigned long passengers_) : name(name_), speed(speed_), passengers(passengers_)
+    Lorry(std::string name_, double speed_) : name(name_), speed(speed_)
     {
-        for(size_t i = 0; i < passengers; ++i)
+        for(size_t i = 0; i < passengerRiskArray.size(); ++i)
         {
             passengerRiskArray.push_back(0);
         }
```

Comment:
`passengerRiskArray.size()` is 0 when this is run.  You still need some initial quantity greater than 0 in order to fill it.
You can keep the constructor parameter `passengers_` and use it here to fill your vector.  Just don't store it as a member variable or use that member variable to keep track of the size of your vector.
- [ ] revise